src/components/shared/Modal.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+
+import { createFactory, DOM as dom, Component, PropTypes } from "react";
+import classnames from "classnames";
+import _Transition from "react-transition-group/Transition";
+const Transition = createFactory(_Transition);
+
+import "./Modal.css";
+
+function Slide({ in: inProp, children }) {
+  return Transition({ in: inProp, timeout: 200 }, children);
+}
+
+export default class Modal extends Component {
+  props: {
+    enabled: boolean,
+    shortcut: string,
+    children: Node,
+    handleOpen: () => any,
+    handleClose: () => any
+  };
+
+  constructor(props) {
+    super(props);
+    const self: any = this;
+    self.onClick = this.onClick.bind(this);
+  }
+
+  componentWillUnmount() {
+    const shortcuts = this.context.shortcuts;
+    shortcuts.off("Escape");
+    shortcuts.off(L10N.getStr(this.props.shortcut));
+  }
+
+  componentDidMount() {
+    const shortcuts = this.context.shortcuts;
+    shortcuts.on("Escape", this.props.handleClose);
+    shortcuts.on(L10N.getStr(this.props.shortcut), this.props.handleOpen);
+  }
+
+  onClick(e: SyntheticEvent) {
+    e.stopPropagation();
+  }
+
+  render() {
+    const { enabled } = this.props;
+    return Slide({
+      in: enabled,
+      children: status => {
+        return dom.div(
```

Comment:
It might help if the wrapper and modal were siblings. This way the modal could animate onto the `<body>`.